StateMachine: Merge Commit and Transfer (#83)
tigerbeetle/tigerbeetle: docs/DEEP_DIVE.md

Again, the second transfer is rejected because it was never created.

```
zig/zig run src/demo_05_post_pending_transfers.zig
zig/zig run src/demo_02_lookup_accounts.zig
```

> Let's also pretend someone else tried to commit (but reject) concurrently:

Reviewer: ```suggestion
Let's also pretend someone else tried to void the pending transfer concurrently:
```

There may be a few more places for the text under the `Demo 4, 5, 6: Two-phase commit journal entries` header, which could also become `Demo 4, 5, 6: Two-phase journal entries`

```
zig/zig run src/demo_06_void_pending_transfers.zig
```

> **From here, feel free to tweak these demos and see what happens. You can explore all our accounting invariants (and the DSL we created for these) in `src/state_machine.zig` by grepping the source for `fn create_account`, `fn create_transfer`, and `fn commit_transfer`.**

Reviewer: ```suggestion
**From here, feel free to tweak these demos and see what happens. You can explore all our accounting invariants (and the DSL we created for these) in `src/state_machine.zig` by grepping the source for `fn create_account` or `fn create_transfer`.**
```
